apidef/oas/xtyk_doc.go

```diff
@@ -0,0 +1,369 @@
+package oas
+
+//go:generate go test -run=TestExtractDocFromXTyk . -timeout 7s -v -x-tyk-dump-doc
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path"
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+type XTykDoc []*StructInfo
+
+func (x *XTykDoc) Append(newInfo *StructInfo) int {
+	*x = append(*x, newInfo)
+	return len(*x)
+}
+
+type StructInfo struct {
+	// Name is struct go name
+	Name string
+	// Fields holds information of the fields, if this object is a struct.
+	Fields    []*FieldInfo    `json:"fields,omitempty"`
+	structObj *ast.StructType `json:"-"`
+}
+
+type FieldInfo struct {
+	// Doc is field docs. comments that are not part of docs are excluded.
+	Doc string `json:"doc"`
+	// JsonName is the corresponding json name of the field.
+	JsonName string `json:"json_name"`
+	// JsonType valid json type if it was found
+	JsonType string `json:"json_type"`
+	// GoPath is the go path of this field starting from root object
+	GoPath string `json:"go_path"`
+	// MapKey is the map key type, if this field is a map
+	MapKey string `json:"map_key,omitempty"`
+	// IsArray reports if this field is an array.
+	IsArray bool `json:"is_array"`
+}
+
+type DocErrList struct {
```

Comment:
apidef/oas/xtyk_doc.go:39: the type name `DocErrList` should conform to the `XxxError` format. (errname)